### ISSUE TITLE
`atuin sync` is not working in MacOS Ventura

### ISSUE THREAD

**Issue by leosuncin** [ellie/atuin#1076]
When I run `atuin sync` the following error appears

```
thread 'main' panicked at 'failed to decrypt history! check your key: could not encrypt

Location:
    atuin-client/src/encryption.rs:125:22', /private/tmp/atuin-20230528-6555-1ryfjaq/atuin-15.0.0/atuin-client/src/api_client.rs:185:44
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

<details>
<summary>RUST_BACKTRACE=1</summary>
<pre>
thread 'main' panicked at 'failed to decrypt history! check your key: could not encrypt

Location:
    atuin-client/src/encryption.rs:125:22', /private/tmp/atuin-20230528-6555-1ryfjaq/atuin-15.0.0/atuin-client/src/api_client.rs:185:44
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
   4: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
   5: atuin_client::sync::sync::{{closure}}
   6: atuin::command::client::Cmd::run::{{closure}}
   7: tokio::runtime::scheduler::current_thread::Context::enter
   8: tokio::macros::scoped_tls::ScopedKey<T>::set
   9: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
  10: tokio::runtime::runtime::Runtime::block_on
  11: atuin::command::client::Cmd::run
  12: atuin::command::AtuinCmd::run
  13: atuin::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
</pre>
</details>

<details>
<summary>RUST_BACKTRACE=full</summary>
<pre>
thread 'main' panicked at 'failed to decrypt history! check your key: could not encrypt

Location:
    atuin-client/src/encryption.rs:125:22', /private/tmp/atuin-20230528-6555-1ryfjaq/atuin-15.0.0/atuin-client/src/api_client.rs:185:44
stack backtrace:
   0:        0x1080a7b46 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h28eefff2fd6348c0
   1:        0x1080c832a - core::fmt::write::h49021f0d249bdcdd
   2:        0x108094fcc - std::io::Write::write_fmt::hfe905b75ea2975e8
   3:        0x1080a792a - std::sys_common::backtrace::print::h729e4781fcd708bf
   4:        0x10808ff23 - std::panicking::default_hook::{{closure}}::he34a760df1040d15
   5:        0x10808fc12 - std::panicking::default_hook::hc21c623951eeae76
   6:        0x108090a95 - std::panicking::rust_panic_with_hook::hf3038b3a4cc27e74
   7:        0x1080a7e64 - std::panicking::begin_panic_handler::{{closure}}::h167648f6debe79bf
   8:        0x1080a7c49 - std::sys_common::backtrace::__rust_end_short_backtrace::h998cad6508c9fd44
   9:        0x10809055d - _rust_begin_unwind
  10:        0x1081044a3 - core::panicking::panic_fmt::h0c7b66c2bc0c7be6
  11:        0x108104915 - core::result::unwrap_failed::h534fb7147885e41d
  12:        0x1077a302f - <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold::h82dd8f04ea2e1300
  13:        0x1077a142e - <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter::ha9b662a29bbab2d0
  14:        0x1077ab73c - atuin_client::sync::sync::{{closure}}::h4b10f61c52c6d443
  15:        0x1077f251c - atuin::command::client::Cmd::run::{{closure}}::hdd42fc7741cb74f9
  16:        0x107850638 - tokio::runtime::scheduler::current_thread::Context::enter::hbbd4cb4790c908f7
  17:        0x107817e92 - tokio::macros::scoped_tls::ScopedKey<T>::set::h2ed0dfedb1e39b1e
  18:        0x10784fb80 - tokio::runtime::scheduler::current_thread::CurrentThread::block_on::h0de5793080a9b2e8
  19:        0x10781c19e - tokio::runtime::runtime::Runtime::block_on::h9df551cc20ae7499
  20:        0x1078b867d - atuin::command::client::Cmd::run::h26eb9d0aa9acf98a
  21:        0x10785c761 - atuin::command::AtuinCmd::run::h2b5d80cda9bdb46c
  22:        0x1077a1ea8 - atuin::main::hd16543e697e3b4a9
  23:        0x107757536 - std::sys_common::backtrace::__rust_begin_short_backtrace::hee8722ce52eeef32
  24:        0x1076e2c91 - std::rt::lang_start::{{closure}}::hb1bb662d5a1a7d8c
  25:        0x10808f224 - std::rt::lang_start_internal::h87ee88830e566709
  26:        0x1077a232c - _main
  27:     0x7ff80dfb241f - <unknown>
</pre>
</details>

I've already tried to logout and login without success, I deleted the key file as mentioned in #613 without luck

**System Information**

OS: MacOS Ventura v13.4.1 x86_64
Atuin: v15.0.0

**Comment by ellie** [comment]
Was this on a second machine, or just the one?

If you have deleted your key file, you will no longer be able to access your account. It is gone.

I'd suggest `atuin account delete`, then register again. Please make a note of your key, and ensure you use it for every new machine you use

Docs for logging in to additional machines: https://atuin.sh/docs/guide#syncing-additional-machines

**Comment by leosuncin** [comment]
Yes, it's on the second machine, I first installed on Linux and sync works, the problem comes in the second machine.

At first it worked out, later I noticed that sync has not been working.

**Comment by leosuncin** [comment]
> Was this on a second machine, or just the one?
> 
> If you have deleted your key file, you will no longer be able to access your account. It is gone.
> 
> I'd suggest `atuin account delete`, then register again. Please make a note of your key, and ensure you use it for every new machine you use
> 
> Docs for logging in to additional machines: https://atuin.sh/docs/guide#syncing-additional-machines

The Mac is from my work, so do I need to delete `~/.local/share/atuin` and start it over?

**Comment by luohy15** [comment]
I also encountered this issue when I use the output of `atuin key` as key of `atuin login`.

Then I tried to use the content of `~/.local/share/atuin/key` as key of `atuin login` and that works

**Comment by conradludgate** [comment]
Can you both confirm the versions of atuin you have on each machine? In newer versions we changed how the key is output (as a mnemonic). 

New versions can still decode the old base64 version, but older versions won't understand the mnemonic.

(if you do see the mnemonic, maybe try `atuin key --base64`)

**Comment by luohy15** [comment]
The atuin version is the same on all my machines: v15.0.0.

In my previous case base64 works but mnemonic not, but it's weird I cannot reproduce it now.

**Comment by ellie** [comment]
In any case where you cannot get sync to work, does copying the `~/.local/share/atuin` directory to a new machine resolve the issue? (just one time, sync should work from then on)

For context: Trying to ensure that the issue lies somehow in how we parse/store keys, and not elsewhere in the process

I've tested

1. Logging out, logging in. No issues
2. Registering a new account, logging out, logging in, adding a second machine to sync. No issues

**Comment by LeoniePhiline** [comment]
@leosuncin Can the issue be closed?
